builder_test.go

```diff
@@ -310,36 +310,286 @@ func testBuilder(t *testing.T, when spec.G, it spec.S) {
+						buildpackStore.EXPECT().Lookup("A", "v1").Return(bpA, nil)
+						bpA.EXPECT().Build(gomock.Any(), config).Return(lifecycle.BuildResult{
+							Processes: []launch.Process{
+								{
+									Type:        "first",
+									Command:     "first-command",
+									Args:        []string{"bpA-arg"},
+									Direct:      true,
+									BuildpackID: "A",
+									Default:     true,
+								},
+								{
+									Type:        "second",
+									Command:     "second-command",
+									Args:        []string{"some-arg"},
+									Direct:      true,
+									BuildpackID: "A",
+									Default:     true,
+								},
+							},
+						}, nil)
+						bpB := testmock.NewMockBuildpack(mockCtrl, builder.Group.Group[1].API)
+						buildpackStore.EXPECT().Lookup("B", "v2").Return(bpB, nil)
+						bpB.EXPECT().Build(gomock.Any(), config).Return(lifecycle.BuildResult{
+							Processes: []launch.Process{
+								{
+									Type:        "third",
+									Command:     "third-command",
+									Args:        []string{"bpB-arg"},
+									Direct:      false,
+									BuildpackID: "B",
+								},
+								{
+									Type:        "fourth",
+									Command:     "fourth-command",
+									Args:        []string{"some-other-arg"},
+									Direct:      true,
+									BuildpackID: "B",
+									Default:     true,
+								},
 							},
+						}, nil)
+
+						metadata, err := builder.Build()
+						if err != nil {
+							t.Fatalf("Unexpected error:\n%s\n", err)
+						}
+
+						if s := cmp.Diff(metadata.Processes, []launch.Process{
 							{
-								Type:        "some-type",
-								Command:     "some-command",
+								Type:        "first",
+								Command:     "first-command",
+								Args:        []string{"bpA-arg"},
+								Direct:      true,
+								BuildpackID: "A",
+								Default:     true,
+							},
+							{
+								Type:        "second",
+								Command:     "second-command",
 								Args:        []string{"some-arg"},
 								Direct:      true,
 								BuildpackID: "A",
+								Default:     true,
+							},
+							{
+								Type:        "third",
+								Command:     "third-command",
+								Args:        []string{"bpB-arg"},
+								Direct:      false,
+								BuildpackID: "B",
+							},
+							{
+								Type:        "fourth",
+								Command:     "fourth-command",
+								Args:        []string{"some-other-arg"},
+								Direct:      true,
+								BuildpackID: "B",
+								Default:     true,
 							},
 						}); s != "" {
 							t.Fatalf("Unexpected:\n%s\n", s)
 						}
 					})
+
+					when("builpack API < 0.6", func() {
+						it("set web processes as default", func() {
+							bpA := testmock.NewMockBuildpack(mockCtrl, builder.Group.Group[0].API)
+							buildpackStore.EXPECT().Lookup("A", "v1").Return(bpA, nil)
+							bpA.EXPECT().Build(gomock.Any(), config).Return(lifecycle.BuildResult{
+								Processes: []launch.Process{
+									{
+										Type:        "some-type",
```

Comment:
Probably we also want a `web` here (bpA is the newer api, right?) to test that we do NOT set it as default if the buildpack doesn't set it.

Reply:
I'll create another test for it. We can't do it in the same test because they'll override one another.

Reply:
Please see https://github.com/buildpacks/lifecycle/pull/520/commits/63f4e9a470ff92f133caa5ffff56d047d6fa3295.